### PR TITLE
Various clue scroll fixes

### PR DESCRIPTION
**Pull request by Rheon-D**
Fixes a few clues

- Added indicator to coordinate clues showing if spade is present in inventory.
- Fixed Item fullfillment failing for `SlotLimitationRequirement`, previously returning false when a slot was unreachable

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollPlugin.java [hunk 1]
```diff
@@ -64,18 +66,7 @@
...
-import net.runelite.client.plugins.cluescrolls.clues.CoordinateClue;
-import net.runelite.client.plugins.cluescrolls.clues.CrypticClue;
-import net.runelite.client.plugins.cluescrolls.clues.EmoteClue;
-import net.runelite.client.plugins.cluescrolls.clues.FairyRingClue;
-import net.runelite.client.plugins.cluescrolls.clues.LocationClueScroll;
-import net.runelite.client.plugins.cluescrolls.clues.MapClue;
-import net.runelite.client.plugins.cluescrolls.clues.NpcClueScroll;
-import net.runelite.client.plugins.cluescrolls.clues.ObjectClueScroll;
-import net.runelite.client.plugins.cluescrolls.clues.TextClueScroll;
+import net.runelite.client.plugins.cluescrolls.clues.*;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollPlugin.java]: Fix your imports: https://github.com/runelite/runelite/wiki/Code-Conventions

File: runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/AnagramClue.java [hunk 2]
```diff
@@ -25,15 +25,16 @@
...
-import java.util.Set;
 import lombok.Getter;
 import net.runelite.api.NPC;
 import net.runelite.api.coords.WorldPoint;
 import net.runelite.client.plugins.cluescrolls.ClueScrollPlugin;
 import net.runelite.client.ui.overlay.OverlayUtil;
 import net.runelite.client.ui.overlay.components.PanelComponent;
+
+import java.awt.*;
+import java.util.Set;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/AnagramClue.java]: Fix your imports https://github.com/runelite/runelite/wiki/Code-Conventions

File: runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/CoordinateClue.java [hunk 3]
```diff
@@ -24,15 +24,18 @@
...
 import lombok.Getter;
 import net.runelite.api.coords.LocalPoint;
 import net.runelite.api.coords.WorldPoint;
 import net.runelite.client.plugins.cluescrolls.ClueScrollPlugin;
+import net.runelite.client.plugins.cluescrolls.clues.emote.ItemRequirement;
+import net.runelite.client.plugins.cluescrolls.clues.emote.SingleItemRequirement;
 import net.runelite.client.ui.overlay.OverlayUtil;
 import net.runelite.client.ui.overlay.components.PanelComponent;
+
+import java.awt.*;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/CoordinateClue.java]: Fix your imports https://github.com/runelite/runelite/wiki/Code-Conventions

File: runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/EmoteClue.java [hunk 4]
```diff
@@ -25,24 +25,19 @@
...
-import net.runelite.client.plugins.cluescrolls.clues.emote.AnyRequirementCollection;
-import net.runelite.client.plugins.cluescrolls.clues.emote.AllRequirementsCollection;
-import net.runelite.client.plugins.cluescrolls.clues.emote.RangeItemRequirement;
-import net.runelite.client.plugins.cluescrolls.clues.emote.SlotLimitationRequirement;
-import net.runelite.client.plugins.cluescrolls.clues.emote.SingleItemRequirement;
+import net.runelite.client.plugins.cluescrolls.clues.emote.*;
 import net.runelite.client.ui.overlay.OverlayUtil;
 import net.runelite.client.ui.overlay.components.PanelComponent;
+
+import java.awt.*;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/EmoteClue.java]: Fix your imports https://github.com/runelite/runelite/wiki/Code-Conventions

File: runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/MapClue.java [hunk 5]
```diff
@@ -25,26 +25,22 @@
...
-import static net.runelite.client.plugins.cluescrolls.ClueScrollWorldOverlay.IMAGE_Z_OFFSET;
-import static net.runelite.client.plugins.cluescrolls.ClueScrollWorldOverlay.SPADE_IMAGE;
 import net.runelite.client.ui.overlay.OverlayUtil;
 import net.runelite.client.ui.overlay.components.PanelComponent;
 
+import java.awt.*;
+import java.util.Set;
+
+import static net.runelite.api.ItemID.*;
+import static net.runelite.api.ObjectID.*;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/MapClue.java]: Fix your imports https://github.com/runelite/runelite/wiki/Code-Conventions

File: runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/EmoteClue.java [hunk 6]
```diff
@@ -123,7 +123,7 @@
...
 		new EmoteClue("Jump for joy in Yanille bank. Dance a jig before you talk to me. Equip a brown apron, adamantite medium helmet and snakeskin chaps.", new WorldPoint(2610, 3092, 0), JUMP_FOR_JOY, JIG, item(BROWN_APRON), item(ADAMANT_MED_HELM), item(SNAKESKIN_CHAPS)),
 		new EmoteClue("Jump for joy in the TzHaar sword shop. Shrug before you talk to me. Equip a Steel longsword, Blue D'hide body and blue mystic gloves.", new WorldPoint(2477, 5146, 0), JUMP_FOR_JOY, SHRUG, item(STEEL_LONGSWORD), item(BLUE_DHIDE_BODY), item(MYSTIC_GLOVES)),
-		new EmoteClue("Jump for joy in the Ancient Cavern. Equip a granite shield, splitbark body, and any rune heraldic helm.", new WorldPoint(2339, 6787, 0), JUMP_FOR_JOY, item(GRANITE_SHIELD), item(SPLITBARK_BODY), range("Any rune heraldic helm", RUNE_HELM_H1, RUNE_HELM_H5)),
+		new EmoteClue("Jump for joy in the Ancient Cavern. Equip a granite shield, splitbark body and any rune heraldic helm.", new WorldPoint(2339, 6787, 0), JUMP_FOR_JOY, item(GRANITE_SHIELD), item(SPLITBARK_BODY), range("Any rune heraldic helm", RUNE_HELM_H1, RUNE_HELM_H5)),
```
Comment (devLotto) [inline runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/EmoteClue.java]: All the different types of heraldic rune helms should be added

File: runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/MapClue.java [hunk 7]
```diff
@@ -67,17 +67,19 @@
...
+		new MapClue(CLUE_SCROLL_HARD_3524, new WorldPoint(2457, 3182, 0), CRATE_18506),
 		new MapClue(CLUE_SCROLL_HARD_3525, new WorldPoint(3026, 3629, 0), CRATE_354),
 		new MapClue(CLUE_SCROLL_HARD_7239, new WorldPoint(3021, 3912, 0)),
 		new MapClue(CLUE_SCROLL_HARD_7241, new WorldPoint(2722, 3338, 0)),
 		new MapClue(CLUE_SCROLL_ELITE_12130, new WorldPoint(2449, 3130, 0)),
 		new MapClue(CLUE_SCROLL_ELITE_19783, new WorldPoint(2202, 3062, 0)),
-		new MapClue(CLUE_SCROLL_ELITE_19784, new WorldPoint(1815, 3852, 0))
+		new MapClue(CLUE_SCROLL_ELITE_19784, new WorldPoint(1815, 3852, 0)),
+		new MapClue(CLUE_SCROLL_ELITE_19784, new WorldPoint(1815, 3852, 0)),
+		new MapClue(CLUE_SCROLL_ELITE_19782, new WorldPoint(2953, 9523, 1))
```
Comment (devLotto) [inline runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/MapClue.java]: Order by item id

File: runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/EmoteClue.java (hunk 6 above)
Comment (devLotto) [inline runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/EmoteClue.java]: All the different types of heraldic rune helms should be added

File: runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/CoordinateClue.java [hunk 8]
```diff
@@ -41,17 +43,24 @@
 {
 	private String text;
 	private WorldPoint location;
+	private static final ItemRequirement HAS_SPADE = new SingleItemRequirement(952);
```
Comment (devLotto) [inline runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/CoordinateClue.java]: This should use ItemID.SPADE, not the constant

File: runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollPlugin.java [hunk 9]
```diff
@@ -335,6 +335,17 @@ public void onGameTick(final GameTick event)
 			}
 		}
 
+		if (clue instanceof CoordinateClue)
```
Comment (devLotto) [inline runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollPlugin.java]: Add ``|| clue instanceof CoordinateClue`` to the ``if (clue instanceof EmoteClue)`` instead

File: runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/EmoteClue.java (hunk 6 above)
Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/EmoteClue.java]: That does include all types of heraldic helm.

```
$ grep -E 'RUNE_HELM_H1 = ' -A4 runelite-api/src/main/java/net/runelite/api/ItemID.java
        public static final int RUNE_HELM_H1 = 10286;
        public static final int RUNE_HELM_H2 = 10288;
        public static final int RUNE_HELM_H3 = 10290;
        public static final int RUNE_HELM_H4 = 10292;
        public static final int RUNE_HELM_H5 = 10294;
```

Comment (devLotto) [reply runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/EmoteClue.java]: Oh, you're right

File: runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollPlugin.java (hunk 9 above)
Comment (Rheon-D) [reply runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollPlugin.java]: The two inventory checks are different, one is checking the equipped items and the other is the actual inventory. I can do it like this if you really want but I think the way I had it is more readable 
```		
boolean isCoordinateClue = clue instanceof CoordinateClue;
		
if (clue instanceof EmoteClue || isCoordinateClue)
{
	ItemContainer container = client.getItemContainer(isCoordinateClue? InventoryID.INVENTORY : InventoryID.EQUIPMENT);
			
	if (container != null)
	{
		equippedItems = container.getItems();
	}
}```

**Comment by deathbeam** [comment]
Needs rebase

**Pull request merged**